### PR TITLE
refactor(themes): update border radius to new logical properties

### PR DESCRIPTION
**Pull request by brandyscarney**
Issue number: internal

---------

## What is the current behavior?
The old `border-radius` mixin added additional selectors and styles depending on whether the direction was set to  `ltr` or `rtl`. 

Old mixin usage:

```scss
.old-border-radius {
  @include border-radius(5px, 6px, 7px, 8px)
}
```

generates:

```css
.old-border-radius {
  border-top-left-radius: 5px;
  border-top-right-radius: 6px;
  border-bottom-right-radius: 7px;
  border-bottom-left-radius: 8px;
}

:host-context([dir=rtl]) .old-border-radius {
  border-top-left-radius: 6px;
  border-top-right-radius: 5px;
  border-bottom-right-radius: 8px;
  border-bottom-left-radius: 7px;
}

[dir=rtl] .old-border-radius {
  border-top-left-radius: 6px;
  border-top-right-radius: 5px;
  border-bottom-right-radius: 8px;
  border-bottom-left-radius: 7px;
}

@supports selector(:dir(rtl)) {
  .old-border-radius:dir(rtl) {
    border-top-left-radius: 6px;
    border-top-right-radius: 5px;
    border-bottom-right-radius: 8px;
    border-bottom-left-radius: 7px;
  }
}
```

## What is the new behavior?
The new `border-radius` mixin uses the [logical properties](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_logical_properties_and_values/Margins_borders_padding) which handles switching based on the [writing-mode](https://developer.mozilla.org/en-US/docs/Web/CSS/writing-mode), [direction](https://developer.mozilla.org/en-US/docs/Web/CSS/direction), and [text-orientation](https://developer.mozilla.org/en-US/docs/Web/CSS/text-orientation):

| Before                       | After                       |
| -----------------------------| ----------------------------|
| `border-top-left-radius`     | `border-start-start-radius` |
| `border-bottom-left-radius`  | `border-end-start-radius`   |
| `border-top-right-radius`    | `border-start-end-radius`   |
| `border-bottom-right-radius` | `border-end-end-radius`     |

New mixin usage:

```scss
.new-border-radius {
  @include border-radius(5px, 6px, 7px, 8px)
}
```

```css
.new-border-radius {
  border-start-start-radius: 5px;
  border-start-end-radius: 6px;
  border-end-end-radius: 7px;
  border-end-start-radius: 8px;
}
```

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

The `select` styles have been updated to use the mixin instead of hardcoding `border-radius`.

### CODE REVIEW

**Review by thetaPC — APPROVED** [review]
LGTM

File: core/src/components/range/range.md.scss [hunk 1]
```diff
@@ -177,11 +177,7 @@
      */
     @include position(null, null, -1px, 50%);
     @include margin-horizontal(-13px, null);
-
-    @include multi-dir() {
-      /* stylelint-disable-next-line property-disallowed-list */
-      border-radius: 50% 50% 50% 0;
-    }
+    @include border-radius(50% 50% 50% 0);
```
Comment (sean-perkins) [inline core/src/components/range/range.md.scss:180]: Is separating arguments in a mixin optional within SASS? For example all our other usages comma separate the individual arguments, but I see here we aren't needing to do that.

Comment (brandyscarney) [reply core/src/components/range/range.md.scss]: This was unintentional! It actually ends up assigning the entire chunk to `border-radius` if you only pass one argument to it:

```scss
.test-border-radius-one-argument {
  @include border-radius(50% 50% 50% 0);
}

.test-border-radius-multiple-arguments {
  @include border-radius(50%, 50%, 50%, 0);
}
```

Becomes:

```css
.test-border-radius-one-argument {
  border-radius: 50% 50% 50% 0;
}

.test-border-radius-multiple-arguments {
  border-start-start-radius: 50%;
  border-start-end-radius: 50%;
  border-end-end-radius: 50%;
  border-end-start-radius: 0;
}
```

I have updated it to be comma separated: https://github.com/ionic-team/ionic-framework/pull/29002/commits/86962bf11bf916f958124b1f56f3625abb2bee52

**Review by sean-perkins — APPROVED** [review]

**Comment by brandyscarney** [comment]
Re-requesting review since I wasn't calling the mixin properly, had to update the code with the correct mixin call.

**Review by thetaPC — APPROVED** [review]
LGTM

**Pull request merged**